yt/data_objects/region_expression.py

```diff
@@ -38,12 +40,19 @@ def __getitem__(self, item):
         if isinstance(item, tuple) and isinstance(item[1], string_types):
             return self.all_data[item]
         if isinstance(item, slice):
-            # This is for the case where we give a slice as an index; one
-            # possible use case of this would be where we supply something
-            # like ds.r[::256j] .  This would be expanded, implicitly into
-            # ds.r[::256j, ::256j, ::256j].  Other cases would be if we do
-            # ds.r[0.1:0.9] where it will be expanded along three dimensions.
-            item = (item, item, item)
+            if iterable(item.start) and len(item.start) == 3 and \
+                iterable(item.stop) and len(item.stop) == 3:
```

Comment:
You might be able to make this logic a little less tortured by defining a helper function:

```
def is_ray_endpoint(obj):
   return iterable(obj) and len(obj) == 3
```